sequence.go

```diff
@@ -660,6 +660,9 @@ func findSequencesOnDisk(path string, opts ...FileOption) (FileSequences, error)
 			padMap[key] = padder.PaddingChars(len(match[2]))
 		} else {
 			frames = append(frames, frame)
+			if p := padder.PaddingChars(len(match[2])); len(padMap[key]) > len(p) {
+				padMap[key] = p
+			}
```

Comment:
This approach compares the length of the padding characters to decide which is the smallest padding character set. I can see a flaw in this approach, and also a bigger flaw even after it is corrected, in the original implementation of finding and sorting the file sequences...

I don't think it is reliable to check the length of padding characters. For instance, `#` and `@` are the same number of characters, but `#` can be 4 and `@` can mean 1. You would need to be checking the actual width of the frame number and finding the width of the number closest to 0. Example:

1000 = `@`
0004 = `#`
003 = `@@@`
2 = `@`
-1 = `@@`

If "0004" is the first path to be evaluated and sets the padding character, than when you get to "2", the length of the padding chars won't change and you wont hit that new branch to update. So that is the initial problem just with the patch. 

"2" has the smallest width in a string representation, and it has the exact same padding character string width as "0004", but -1 is closer to 0 and it needs 2-pad (because of the negative). 

The remanding problem, even after we properly determine the padding for the sequence, is that we are only considering the key to be the basename and extension. But what if we have the following?

base.1.ext
base.02.ext
base.003.ext

I think the current implementation would return that as one sequence. The existing code would be whatever padding was evaluated first (which ever frame the file system returned first). With your patch, we would get back one sequence with a padding of `@`. But you can see that this is broken.    "base.1-3@.ext" is not correct as it would not find the second and third frames with that padding. This should really be 3 results. We would need to make the key `[3]string{base, padChars, ext}`. 

I'm still left with a question though. If we keyed on the padding chars as well, what happens in this circumstance...

base.1.ext
base.02.ext
base.1000.ext
base.1001.ext

"base.1000.ext" could fit into any padding that is 4 or less. So should this return 2 or 3 sequences?

base.1,1000-1001@.ext
base.2,1000-1001@@.ext
base.1000-1001#.ext (or base.1000-1001@.ext ?)

I'm not sure what to do in the case where there is a mixed padding. The one thing I am sure of is that it doesn't make sense to return any obviously wrong patterns that result in files not being found.